Geometry/HGCalCommonData/src/HGCalGeomParameters.cc

```diff
@@ -70,10 +70,20 @@ void HGCalGeomParameters::loadGeometryHexagon(const DDFilteredView& _fv,
           php.layer_.end()) php.layer_.emplace_back(lay);
       auto itr = layers.find(lay);
       if (itr == layers.end()) {
-        const DDTubs & tube = static_cast<DDTubs>(sol);
-        double rin = HGCalParameters::k_ScaleFromDDD*tube.rIn();
-        double rout= HGCalParameters::k_ScaleFromDDD*tube.rOut();
-        double zp  = HGCalParameters::k_ScaleFromDDD*fv.translation().Z();
+	double rin(0), rout(0);
+	if ((sol.shape() == DDSolidShape::ddpolyhedra_rz) ||
+	    (sol.shape() == DDSolidShape::ddpolyhedra_rrz)) {
+	  const DDPolyhedra& polyhedra = static_cast<DDPolyhedra>(sol);
+	  std::vector<double> rmin = polyhedra.rMinVec();
```

Comment:
`const std::vector<double>& rmin` to avoid an unnecessary copy (also for the next one)

Reply:
done